proposals/readonly-struct-init.md

```diff
@@ -0,0 +1,161 @@
+completed. That is once the code calling `new` has a reference to the value there should be no 
+further modification of `readonly` fields. Until the calling code has a reference to the object
+modifications can continue to be made via `this`.
+
+This makes it even easier to define usable `readonly struct` types in C#. The simplest form of
+declaring the fields now creates a fully usable value:
+
+``` csharp
+readonly struct Point { 
+    readonly int X;
+    readonly int Y; 
+}
+
+var p = new Point { X = 42; Y = 13 };
+```
+
+This means authoring `readonly struct` types in C# is just as easy as creating the less
+perferred mutable `struct` types. There is no longer a need to define a constructor which simply 
+re-lists every field as a parameter and then assigns them one at a time. The definition of the 
+field itself allows provides a mechanism for initialization
+
+## Detailed Design 
+
+### Language
+
+The language will extend the object initialization rules to allow for an accessible readonly field 
+or auto-implemented property to appear in the member initialization list. In such a context the 
+members will allow for assignment just as it would inside a constructor of the object. 
+
+The following restrictions will apply to this extension:
+
+1. The type being instantiated must be a `struct` or `ref struct`. 
+2. The constructor used in the initializer cannot have any `out` or `ref` parameters.
+
+The (2) restriction exists because such a parameter could smuggle out `this` from the object 
+constructor. That would make a reference to the constructed object visible to the code calling
+`new` and hence allow it to observe mutations to the value.
+
+### Compiler 
+
+The compiler will change how the backing field of an auto-implemented property is emitted in the 
+following ways:
+
+- The accessibility of the backing field will match the accessibility of the property.
+- The name of the field will be unspeakable (illegal C# identifier hence never imported) but it 
+will be possible to map such fields to the properties that defined them.
+
+Taken together this allows the C# compiler to translate a get only auto-implemented property into
+a simple field initialization when used as an object initializer. 
+
+``` csharp 
+struct Name {
+    // Backing field: public readonly string <FirstName>__BackingField;
+    public string FirstName { get; }
+    // Backing field: public readonly string <LastName>__BackingField;
+    public string LastNmae { get; }
+}
+
+var name = new Name { FirstName = "Jared", LastName = "Parsons" };
+
+// Emitted as 
+var name = new Name { <FirstName>__BackingField = "Jared", <LastName>__BackingField = "Parsons" };
+```
+
+This does mean that the feature will only apply to auto-implemented properties that are 
+re-compiled with a new compiler. The compiler today emits the fields as `private readonly` and 
+hence will be inaccessible to calling code. Upon recompile the backing fields will be emitted 
+as specified above.
```

Comment:
This will also have impact on the tools that reason about public surface (e.g. tooling used by the framework and documentation teams).